src/java/org/apache/cassandra/cql3/restrictions/IndexRestrictions.java

```diff
@@ -63,6 +65,49 @@ public List<CustomIndexExpression> getCustomIndexExpressions()
         return customExpressions;
     }
 
+    /**
+     * Returns whether these restrictions would need filtering if the specified index registry were used.
+     *
+     * @param indexRegistry an index registry
+     * @return {@code true} if this would need filtering if {@code indexRegistry} were used, {@code false} otherwise
+     */
+    public boolean needFiltering(IndexRegistry indexRegistry)
```

Comment:
need_S_Filtering?

Reply:
I wasn't sure about this one. If you read this as "these index restrictions need filtering" then it's plural. However, if you read this as "this instance of the `IndexRestrictions` class needs filtering" then it's 3rd person singular. I have changed it to the second criterion, using `needsFiltering`.